src/utils/hex.ts

```diff
@@ -0,0 +1,79 @@
+import { BrandedString } from '../types'
+
+/**
+ * Nominal type to represent hex strings
+ */
+export type HexString = BrandedString<'HexString'>
+
+/**
+ * Strips the '0x' hex prefix from a string
+
+ * @param hex string input
+ */
+export function stripHexPrefix<T extends string>(hex: T): T {
+  return hex.startsWith('0x') ? (hex.slice(2) as T) : hex
+}
+
+/**
+ * Converts a hex string to array of numbers
+ *
+ * @param hex string input
+ */
+export function hexToByteArray(hex: HexString): number[] {
+  const hexWithoutPrefix = stripHexPrefix(hex)
+  const subStrings: string[] = []
+  for (let i = 0; i < hexWithoutPrefix.length; i += 2) {
+    subStrings.push(hexWithoutPrefix.substr(i, 2))
+  }
+
+  return subStrings.map(s => parseInt(s, 16))
```

Comment:
```suggestion
  return hexWithoutPrefix.match(/.{1,2}/g).map(byte => parseInt(byte, 16))
```

Reply:
@vojtechsimetka thanks for the suggestion.
Can I ask if we use any regex, can we put a comment above the codeline what exactly it does in order to human earthly mortals are able to understand and read immediately like me? : D
seems more effective, although I could read @agazso's code better.

Reply:
Sure. This one just splits the input into array an of 2 character long strings.

`.` matches any character except for line terminator
`{1,2}` quantifier does a greedy match of maximum 2 consequentive characters as long as there are two. If the total length is odd, it also returns the last character (that is why the minimum in quantifier is `1`)

e.g.
```
'1234' =>  ['12', '34'] 
'12345' =>  ['12', '34', '5']
```

Reply:
cool, I checked it, I'm just saying we should comment it above the codeline in the source code.

Reply:
> seems more effective, although I could read @agazso's code better.

I just checked and it's actually not more efficient, I actually just found it way more readable. 

So let's keep @agazso code. If we do want to optimize (eliminate one cycle) we can do the `parseInt` right away

```suggestion
  const subStrings: string[] = []
  for (let i = 0; i < hexWithoutPrefix.length; i += 2) {
    subStrings.push(parseInt(hexWithoutPrefix.substr(i, 2), 16))
  }

  return subStrings
```

Reply:
I like this latest version